private_gpt/components/embedding/embedding_component.py

```diff
@@ -28,9 +28,33 @@ def __init__(self, settings: Settings) -> None:
                         "Local dependencies not found, install with `poetry install --extras embeddings-huggingface`"
                     ) from e
 
+                # Get the number of available GPUs
+                num_gpus = torch.cuda.device_count()
```

Comment:
Adding code to the codebase just to print information is not a good practive. I'd remove this. whole block of prints.